superset/sql_parse.py

```diff
@@ -752,11 +752,19 @@ def _extract_tables_from_sql(self) -> set[Table]:
             statements = parse(self.stripped(), dialect=self._dialect)
         except SqlglotError as ex:
             logger.warning("Unable to parse SQL (%s): %s", self._dialect, self.sql)
-            dialect = self._dialect or "generic"
+
+            if isinstance(ex, ParseError):
+                error = ex.errors[0]
+                suffix = f"""Error parsing near '{error["highlight"]}' at line {error["line"]}:{error["col"]}"""  # pylint: disable=line-too-long
```

Comment:
Could simplify to:

```suggestion
                suffix = "Error parsing near '{highlight}' at line {line}:{col}".format(**error)
```

Thought the linter usually complain of using `.format()`.